packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -725,3 +725,108 @@ class _MotionEventsDispatcher {
   bool isSinglePointerAction(PointerEvent event) =>
       !(event is PointerDownEvent) && !(event is PointerUpEvent);
 }
+
+/// A callback used by [PlatformViewRenderBox] to create a layer.
+///
+/// Implement this if you need to ask [PlatformViewRenderBox] to create a custom layer instead of a standard [PlatformViewLayer].
+/// The `renderBox` is the render object whose `paint` method is invoked to create this layer.
+/// The `context` and `offset` parameters are the same as what will be passed to the [RenderBox.paint] method.
+/// The `id` is the platform view id provided to you by the framework.
+typedef PlatformViewRenderBoxLayerBuilder = Layer Function(PlatformViewRenderBox renderBox, PaintingContext context, Offset offset, int id);
```

Comment:
Do we have a concrete use case for this?
I'd imaging that a platform that's using a layer other than PlatformViewLayer will need a different render object anyway.

Reply:
maybe i'll remove this and add it back when/if we have a concrete case?